index.js

```diff
@@ -1039,6 +1039,62 @@
       [Fn(a, $.Boolean), m(a), m(a)],
       Z.filterM);
 
+  function Array$findIndex(pred, xs) {
+    for (var idx = 0; idx < xs.length && pred(xs[idx]); idx += 1);
+    return idx;
+  }
```

Comment:
My mistake. Even returning `-1` it would still be incorrect. I was naming it `findIndex` but it would really have been `findLastIndex` if returning `-1` for failure.

I'll make `Array$takeWhile` and `Array$dropWhile` instead.